src/set.c

```diff
@@ -523,7 +523,7 @@ static Obj FuncREM_SET(Obj self, Obj set, Obj obj)
     if ( pos <= len && EQ( ELM_PLIST(set,pos), obj ) ) {
 
         Obj * ptr = ADDR_OBJ(set) + pos;
-        SyMemmove(ptr, ptr + 1, sizeof(Obj) * (len - pos + 1));
+        SyMemmove(ptr, ptr + 1, sizeof(Obj) * (len - pos));
```

Comment:
To illustrate why this is right (and was wrong before): If `pos == len`, then we are removing the last element, and 0 bytes need to be moved. If `pos == len-1`, then we need to move `len - pos == 1` object pointers; and so on.